src/ability.js

```diff
@@ -321,13 +322,22 @@ export class Ability {
 	 * @return {void}
 	 */
 	animation2(o) {
-		let game = this.game,
-			opt = $j.extend({ callback: function () {}, arg: {} }, o),
-			args = opt.arg,
-			activateAbility = () => {
-				this.activate(args[0], args[1]);
-				this.postActivate();
-			};
+		const game = this.game;
+		const opt = $j.extend(
+			{
+				callback: function () {
+					// Default no-op function.
+				},
+				arg: {},
+			},
+			o,
+		);
+		const args = opt.arg;
+		const activateAbility = () => {
+			const extra = args[2];
+			this.activate(args[0], args[1], extra);
```

Comment:
This is the real change, allowing us to pass the original `query()` options through to the animation and into the `active()` function.